jlm/llvm/opt/cne.cpp

```diff
@@ -479,29 +476,27 @@ divert(rvsdg::Region *, cnectx &);
 static void
 divert_gamma(rvsdg::StructuralNode * node, cnectx & ctx)
 {
-  JLM_ASSERT(rvsdg::is<rvsdg::GammaOperation>(node));
-  auto gamma = static_cast<rvsdg::GammaNode *>(node);
+  auto & gamma = dynamic_cast<rvsdg::GammaNode &>(*node);
 
-  for (const auto & ev : gamma->GetEntryVars())
+  for (const auto & ev : gamma.GetEntryVars())
   {
     for (auto input : ev.branchArgument)
       divert_users(input, ctx);
   }
 
-  for (size_t r = 0; r < node->nsubregions(); r++)
-    divert(node->subregion(r), ctx);
+  for (size_t r = 0; r < gamma.nsubregions(); r++)
+    divert(gamma.subregion(r), ctx);
 
   divert_outputs(node, ctx);
 }
 
 static void
 divert_theta(rvsdg::StructuralNode * node, cnectx & ctx)
 {
-  JLM_ASSERT(is<rvsdg::ThetaOperation>(node));
-  auto theta = static_cast<rvsdg::ThetaNode *>(node);
-  auto subregion = node->subregion(0);
+  auto & theta = dynamic_cast<rvsdg::ThetaNode &>(*node);
+  auto subregion = theta.subregion();
 
-  for (const auto & lv : theta->GetLoopVars())
+  for (const auto & lv : theta.GetLoopVars())
```

Comment:
same here and the other changes further down

Reply:
all reverted